streaming/base/stream.py

```diff
@@ -319,3 +407,27 @@ def get_shards(self, world: World) -> List[Reader]:
             shards.append(shard)
 
         return shards
+
+    def init_local_dir(self, shards: List[Reader]) -> List[bool]:
+        """Bring a local directory into a consistent state, getting which shards are present.
+
+        Args:
+            shards (List[Reader]): List of this stream's shards.
+
+        Returns:
+            List[bool]: List of whether each stream shard is present.
+        """
+        # List the cache directory (so that we hit the filesystem once).
+        local_dirname = os.path.join(self.local, self.split)
+        ls = set()
+        for dirname, _, subfiles in os.walk(local_dirname):
```

Comment:
are subdirs explicitly disallowed somewhere? otherwise this would just silently skip them, yeah?

Reply:
yeah we don't care about dirs, just shards